src/NUnitTestExtractor/NUnitTestExtractor.Tests/NUnitTestExtractorTests.cs

```diff
@@ -3,22 +3,309 @@
     public class NUnitTestExtractorTests
     {
-        [Test]
-        public void DirectoryCreatedWhenOutput()
+        private const string outputDirectory = "C:/Git/NTestController/src/NUnitTestExtractor/NUnitTestExtractor.Tests/bin/Debug/TestOutput/file.txt";
+        
+        /// <summary>
+        /// Used to invoke the GetTests() method in NUnitTestExtractorApp
+        /// </summary>
+        /// <param name="dlls">The dlls to pass to GetTests()</param>
+        /// <param name="level">The level to pass to GetTests()</param>
+        private void InvokeGetTestsUsingFileOutput(List<string> dlls, NUnitTestExtractorApp.Level level)
         {
+            using (StreamWriter writer = new StreamWriter(outputDirectory))
+            {
+                NUnitTestExtractorApp.GetTests(dlls, level, writer);
+            }
+        }
+
+        /// <summary>
+        /// Sets up and runs process which is used to read lines in the standard output
+        /// </summary>
+        /// <param name="dll">Dlls to pass to the process</param>
+        /// <param name="level"> level to pass to the process</param>
+        /// <returns></returns>
+       private Process SetUpProcess(string dll, string level)
+        {
+            Process process = new Process();
+
+            process.StartInfo.FileName = "NUnitTestExtractor.exe";
+            process.StartInfo.Arguments = "-l " + level + " " + dll;
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+
+            process.Start();
+
+            return process;
+        }
+
+        [SetUp]
+        public void CreateDirectoryAndFile()
+        {
+            int index = outputDirectory.LastIndexOf("/", StringComparison.OrdinalIgnoreCase);
+
+            string directory = "";
+
+            if (index > 0)
+            {
+                directory = outputDirectory.Substring(0, index);
+            }
+
+            if (directory != null && !Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+            }
+
+            if (!File.Exists(outputDirectory))
+            {
+                using (FileStream stream = File.Create(outputDirectory)) { }
+            }
+        } 
+
+        [TearDown]
+        public void EraseOutputFile()
+        {
+            if (File.Exists(outputDirectory))
+            {
+                File.Delete(outputDirectory);
+            }
+        }
+
+        [TestCaseSource(typeof(InvalidDataTestCases))]
+        public void WritingToFile_InvalidDll_NothingWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            using (StreamReader reader = new StreamReader(outputDirectory))
+            {
+                string lines = reader.ReadToEnd();
+
+                Assert.That(lines, Is.Empty);
```

Comment:
Although the NUnit Asserts don't require an assert message (I wish they made it required) we should add a message describing why the test is failing.